packages/container/lib/container.js

```diff
@@ -527,25 +517,27 @@ function factoryInjectionsFor(container, fullName) {
   return factoryInjections;
 }
 
-// TODO - remove when Ember reaches v3.0.0
-function injectDeprecatedContainer(object, container) {
-  if ('container' in object) { return; }
-  Object.defineProperty(object, 'container', {
-    configurable: true,
-    enumerable: false,
-    get() {
-      deprecate('Using the injected `container` is deprecated. Please use the `getOwner` helper instead to access the owner of this object.', false, { id: 'ember-application.injected-container', until: '3.0.0', url: 'http://emberjs.com/deprecations/v2.x#toc_injected-container-access' });
-      return this[CONTAINER_OVERRIDE] || container;
-    },
+const INJECTED_DEPRECATED_CONTAINER_DESC = {
+  configurable: true,
+  enumerable: false,
+  get() {
+    deprecate('Using the injected `container` is deprecated. Please use the `getOwner` helper instead to access the owner of this object.', false, { id: 'ember-application.injected-container', until: '3.0.0', url: 'http://emberjs.com/deprecations/v2.x#toc_injected-container-access' });
+    return this[CONTAINER_OVERRIDE] || getOwner(this);
+  },
 
-    set(value) {
-      deprecate(`Providing the \`container\` property to ${this} is deprecated. Please use \`Ember.setOwner\` or \`owner.ownerInjection()\` instead to provide an owner to the instance being created.`, false, { id: 'ember-application.injected-container', until: '3.0.0', url: 'http://emberjs.com/deprecations/v2.x#toc_injected-container-access' });
+  set(value) {
+    deprecate(`Providing the \`container\` property to ${this} is deprecated. Please use \`Ember.setOwner\` or \`owner.ownerInjection()\` instead to provide an owner to the instance being created.`, false, { id: 'ember-application.injected-container', until: '3.0.0', url: 'http://emberjs.com/deprecations/v2.x#toc_injected-container-access' });
 
-      this[CONTAINER_OVERRIDE] = value;
+    this[CONTAINER_OVERRIDE] = value;
 
-      return value;
-    }
-  });
+    return value;
+  }
+};
+
+// TODO - remove when Ember reaches v3.0.0
+function injectDeprecatedContainer(object) {
+  if ('container' in object) { return; }
+  Object.defineProperty(object, 'container', INJECTED_DEPRECATED_CONTAINER_DESC);
```

Comment:
looks like `container.owner` is null for the offending tests. and this PR closing over `container` masked that.

@rwjblue do i expect `container.owner` to be null? Is that a test config error?